### DISCUSSION TITLE
Allow for filtering on spread embedded resources

### DISCUSSION THREAD

**Discussion by stefan-girlich** [supabase/supabase#26607] [Feature Requests] [answered]
### Scenario
- Each order has one article
- I want to find all orders where an article with a certain minimum value has been ordered
- supabase-js 2.43.2

### Current behavior
```ts
supabase
    .from('orders')
    .select('order_id, ...articles!inner(*)')
    .gte('price', 123)
```

=> `Error: column orders.price does not exist`

### Desired behavior
When using the spread operator, I can filter by the spread properties as if they were part of the original top-level resource.

**Comment by dule-kitemetric** [comment]
It would be great if Supabase supports this, then I don't have to create computed fields/views for filtering source table by nested referenced tables

**Reply by stefan-girlich** [reply, marked as answer]
@dule-kitemetric You can use the embedded table name as a prefix:

```ts
await supabase
    .from('users')
    .select('id, ...organizations!inner(name)')
    .eq('organizations.name', 'A specific organization')
```
```js
[
  {
    id: '01931fbb-df5a-0325-a8ec-590cfdda376f',
    name: 'A specific organization'
  },
  /* ... */
]
```

Noe that without `!inner` all `users` rows are returned with `name: null` where the filter does not apply.

**Reply by dule-kitemetric** [reply]
@stefan-girlich but how about complex query? (e.g. OR condition)
I cannot query like these to filter source table by referenced table
- Syntax error
``` TS
await supabase
    .from('users')
    .select('id, ...organizations!inner(name, code)')
    .or('organizations.name.eq.organization_name', 'organizations.code.eq.organization_code')
   ```
- This also doesn't work
``` TS
await supabase
    .from('users')
    .select('id, ...organizations!inner(name, code)')
    .or('name.eq.organization_name', 'code.eq.organization_code')
   ```

**Reply by stefan-girlich** [reply]
`organization_name` and `organization_code` are not fields in `select()`. Would `.or('organizations.name.eq.name', 'organizations.code.eq.code')` work?
